blocks/library/embed/index.js

```diff
@@ -286,6 +337,7 @@ export const common = [
 			title: 'Twitter',
 			icon: 'embed-post',
 			keywords: [ __( 'tweet' ) ],
+			patterns: [ '(www\.)?twitter\.com\/.+' ],
```

Comment:
Why do we have regexes here? Why can't we discover these from the embed API?

The provider name comes from the oembed response, so if you look at the list of providers in `wp-includes/class-oembed.php` you'll see it's just an array of regexes and oembed URLs. There's nothing there to tie each provider to a block.

So these patterns are used to match a URL to a block. They're not the full regex used by oembed, they just match enough of the URL to be able to match it to the block.